sunpy/net/fido_factory.py

```diff
@@ -385,10 +385,23 @@ def fetch(self, *query_results, path=None, max_conn=5, progress=True,
         if "wait" in kwargs:
             raise ValueError("wait is not a valid keyword argument to Fido.fetch.")
 
+        # TODO: Hack for JSOC to avoid more than one connection.
+        # Remove when parfive allows us to special case URLS.
+        is_jsoc_only = False
+        for query_result in query_results:
+            if isinstance(query_result, UnifiedResponse):
+                is_jsoc_only = all([result.client.__class__.__name__ ==
+                                    "JSOCClient" for result in query_result])
```

Comment:
Why not use `isinstance` or `issubclass` here?

Reply:
I don't know.  That makes more sense.